include/opae/access.h

```diff
@@ -72,6 +72,35 @@ extern "C" {
 fpga_result fpgaOpen(fpga_token token, fpga_handle *handle,
 		     int flags);
 
+/**
+ * Extract the handles of children of a previously fpgaOpen()ed resource.
+ * Only AFUs with feature parameters that name child AFU GUIDs will have
+ * children.
+ *
+ * Child AFU handles may be used to connect to child-specific MMIO regions.
+ * There is no need to close child handles with fpgaClose(). The handles
+ * will be closed automatically when the parent is closed. Child handles
+ * may not be passed to fpgaPrepareBuffer(). All shared memory management
+ * must be associated with the parent.
+ *
+ * @param[in]  handle       Handle to previously opened FPGA object
+ * @param[in]  max_children Maximum number of handles that may be returned
+ *                          in the `children` array.
+ * @param[out] children     Pointer to an array of child handles currently
+ *                          open. When NULL or `max_children` is 0, the
+ *                          number of children will still be returned in
+ *                          `num_children`.
+ * @param[out] num_children Number of children belonging to the parent.
+ *                          This number may be higher than `max_children`.
+ * @returns                 FPGA_OK on success.
```

Comment:
If child AFU GUID not available or present,  fpgaGetChildren() going to return FPGA_NOT_FOUND ?  
fpga_handle (OPAE FPGA resources Handle ) is FPGA_ACCELERATOR type ?

Reply:
No, fpgaOpen() will fail. I actually didn't want to expose child handles at all, since they are just needed for MMIO. Is there something better we should do? The only thing I could think of required adding a child ID parameter to all the MMIO methods, which seems like a bad idea.

Assuming we go with fpgaGetChildren(), what do you think about the problem of wrapping child handles and releasing them automatically? Is that possible?